tests/virtctl/memorydump.go

```diff
@@ -345,3 +349,11 @@ func verifyMemoryDumpFile(dumpFilePath, dumpName string) {
 	Expect(err).ToNot(HaveOccurred())
 	Expect(stat.Size()).To(BeNumerically(">", 0))
 }
+
+func sanitizedPath(extractPath, name string) (string, error) {
+	path := filepath.Join(extractPath, name)
+	if !strings.HasPrefix(path, filepath.Clean(extractPath)) {
```

Comment:
Shouldn't we sanitize here the entire path? What if `extractPath` is a proper path but `path` is something like `../../path`

Reply:
`filepath.Join` will already clean `path`, so both values passed to `strings.HasPrefix` are cleaned. Is there anything else we could do?

Reply:
See https://pkg.go.dev/path/filepath#Join

Reply:
Isn't then `strings.HasPrefix(path, filepath.Clean(extractPath))` always true?

Reply:
And above I meant `name`  not path

Reply:
No, e.g. if `name` has leading double dots then `path` will not have `extractPath` as prefix.

```
extractPath := /test/path
name := ../testfile

path := filepath.Join(extractPath, name) --> /test/testfile

strings.HasPrefix(path, filepath.Clean(extractPath)) --> false
```

Therefore it should be correct to clean `extractPath` and `name` after they have been joined.

Reply:
Ah, I see